### PR TITLE
Crete new error in case that promise is rejected

### PR DESCRIPTION
**Pull request by ronyad**

### CODE REVIEW

File: lib/Worker.js [hunk 1]
```diff
@@ -74,8 +74,8 @@ module.exports = (function(){
                             return liveEntry;
                         })
                         .catch(function(err) {
-                            err.entry = liveEntry;
                             logger.debug("Live entry [%s] Failed to start", e.entryId);
```
Comment (kalturaguy) [inline lib/Worker.js]: it was helpfull if you would print err....

File: lib/entry/LiveEntry.js [hunk 2]
```diff
@@ -138,7 +138,8 @@ LiveEntry.prototype.stop = function() {
                 });
         })
         .catch(function() {
-            return Q.reject();
+            //todo we need this line? ron 
+            return Q.reject(new Error("Failed to stop promise"));
```
Comment (kalturaguy) [inline lib/entry/LiveEntry.js]: why not pass the error in the catch?

File: lib/utils/promise-utils.js [hunk 3]
```diff
@@ -37,7 +37,7 @@ function SynchornizedPromises(logger) {
                         deferred.resolve(result[0].value);
                     }
                     else {
-                        deferred.reject(result[0].reason);
+                        deferred.reject(new Error ("result[0].reason"));
```
Comment (kalturaguy) [inline lib/utils/promise-utils.js]: deferred.reject(result[0].reason); will reject the actual error

File: lib/utils/error-utils.js [hunk 4]
```diff
@@ -8,6 +8,9 @@ var ErrorUtils = (function(){
 
     var obj = {};
 
+    obj.printError=function (err) {
```
Comment (kalturaguy) [inline lib/utils/error-utils.js]: rename to error2string

File: lib/Worker.js [hunk 5]
```diff
@@ -74,8 +75,10 @@ module.exports = (function(){
                             return liveEntry;
                         })
                         .catch(function(err) {
-                            err.entry = liveEntry;
-                            logger.debug("Live entry [%s] Failed to start", e.entryId);
+                            logger.error("Live entry [%s] Failed to start", e.entryId);
+                            if (_.isObject(err)){
```
Comment (gadyaari) [inline lib/Worker.js]: Please move [%s] to beginning of string to be consistent with other logger printings
,

**Pull request merged**